pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,73 @@
+import NextAuth from "next-auth";
+import Providers from "next-auth/providers";
+import { PrismaClient, users } from "@prisma/client";
+import { createHmac } from "crypto";
+import { NextApiRequest, NextApiResponse } from "next";
+
+type AuthorizeDTO = {
+  email: string;
+  password: string;
+  inviteCode?: string;
+  // metadata: userData;
```

Comment:
Nit: we'll let @christopher-grey and @shannonmhong take this one. We can drop this comment.
```suggestion
```